### ISSUE TITLE
Exception thrown instead of null when passing unknown material into matchMaterial()

### ISSUE THREAD

**Issue by ghost** [PaperMC/Paper#2963]
### What behaviour is expected:
What should have happened is it return null on matchMaterial() instead of showing an exception. Both the getMaterial() and matchMaterial() methods are `@Nullable` and have no `throws`. 

### What behaviour is observed:
An exception is thrown when an unknown material is passed into matchMaterial(), this was NOT seen until 1.15.

### Steps/models to reproduce:
`Material.matchMaterial("SIGN", true);`
Note that we are passing in a material that no longer exists in 1.15 (it is now OAK_SIGN).
This should return null but instead it throws an exception.

### Plugin list:
[17:55:15 INFO]: Plugins (22): AuthMe, BungeeTabListPlus, ChestShop, CoreProtect, Factions, FactionsTop, GriefPrevention, Holograms, mcMMO, Multiverse-Core, Multiverse-NetherPortals, PlotSquared, ProtocolLib, ProtocolSupport, ProtocolSupportLegacySupport, SkorrloreGaming, SkorrloreGamingLink, uSkyBlock, Vault, Votifier, WorldEdit, WorldGuard

### Paper build number:
[17:55:27 INFO]: This server is running Paper version git-Paper-105 (MC: 1.15.2) (Implementing API version 1.15.2-R0.1-SNAPSHOT)
[17:55:27 INFO]: Checking version, please wait...
[17:55:28 INFO]: Previous version: git-Paper-23 (MC: 1.15.1)
[17:55:28 INFO]: You are running the latest version

### Anything else:
```Java
[17:40:05 INFO]: [uSkyBlock] Enabling uSkyBlock v2.8.8-SNAPSHOT
[17:40:05 WARN]: Initializing Legacy Material Support. Unless you have legacy plugins and/or data this is a bug!
[17:40:05 WARN]: java.lang.Exception
[17:40:05 WARN]: at org.bukkit.craftbukkit.v1_15_R1.legacy.CraftLegacy.<clinit>(CraftLegacy.java:259)
[17:40:05 WARN]: at org.bukkit.craftbukkit.v1_15_R1.util.CraftMagicNumbers.fromLegacy(CraftMagicNumbers.java:148)
[17:40:05 WARN]: at org.bukkit.Material.getMaterial(Material.java:4265)
[17:40:05 WARN]: at org.bukkit.Material.matchMaterial(Material.java:4310)
[17:40:05 WARN]: at us.talabrek.ultimateskyblock.utils.util.ItemStackUtil.getItemType(ItemStackUtil.java:62)
[17:40:05 WARN]: at us.talabrek.ultimateskyblock.utils.util.ItemStackUtil.createItemStackAmount(ItemStackUtil.java:95)
[17:40:05 WARN]: at us.talabrek.ultimateskyblock.utils.util.ItemStackUtil.createItemList(ItemStackUtil.java:82)
[17:40:05 WARN]: at us.talabrek.ultimateskyblock.Settings.loadPluginConfig(Settings.java:111)
[17:40:05 WARN]: at us.talabrek.ultimateskyblock.uSkyBlock.reloadConfigs(uSkyBlock.java:782)
[17:40:05 WARN]: at us.talabrek.ultimateskyblock.uSkyBlock.onEnable(uSkyBlock.java:209)
[17:40:05 WARN]: at org.bukkit.plugin.java.JavaPlugin.setEnabled(JavaPlugin.java:263)
[17:40:05 WARN]: at org.bukkit.plugin.java.JavaPluginLoader.enablePlugin(JavaPluginLoader.java:337)
[17:40:05 WARN]: at org.bukkit.plugin.SimplePluginManager.enablePlugin(SimplePluginManager.java:435)
[17:40:05 WARN]: at org.bukkit.craftbukkit.v1_15_R1.CraftServer.enablePlugin(CraftServer.java:470)
[17:40:05 WARN]: at org.bukkit.craftbukkit.v1_15_R1.CraftServer.enablePlugins(CraftServer.java:384)
[17:40:05 WARN]: at net.minecraft.server.v1_15_R1.DedicatedServer.init(DedicatedServer.java:235)
[17:40:05 WARN]: at net.minecraft.server.v1_15_R1.MinecraftServer.run(MinecraftServer.java:885)
[17:40:05 WARN]: at java.lang.Thread.run(Thread.java:748)
```

Full log [https://pastebin.com/raw/xWxb7Vca](https://pastebin.com/raw/xWxb7Vca)

**Comment by wizjany** [comment]
What's the behavior on spigot/craftbukkit?

**Comment by ghost** [comment]
Uh, give me a moment, my server does not work on Spigot at all so let me get a test server running.

**Comment by ghost** [comment]
@wizjany There is no such exception on Spigot, it just shows initializing legacy material support as it should. This is paper specific it seems.

**Comment by BillyGalbreath** [comment]
Please update your server. This was fixed a long time ago by moving the exception down to debug level instead of warn.

**Comment by ghost** [comment]
@BillyGalbreath I am using the latest build 
edit: but thanks for letting me know that I have to turn off debug to hide it
edit 2: disabling debug fixed it, thanks, but now I have to enable debug everytime I need to fix something in 1.7 again LOL. I guess I have to weigh my options, have an exception that shouldn't be there to begin with, even with debug on, and be able to debug my plugins or not have the exception and not be able to debug my plugins

**Comment by wizjany** [comment]
It's a stacktrace. It's not even throwing anything.

**Comment by BillyGalbreath** [comment]
> ... but now I have to enable debug everytime I need to fix something in 1.7 again LOL. I guess I have to weigh my options ...

Oooooorrrrrrr, just stop using legacy b.s. and let it die.

**Comment by ghost** [comment]
> 
> 
> It's a stacktrace. It's not even throwing anything.

Yeah I see that, bad choice of words.

> 
> 
> > ... but now I have to enable debug everytime I need to fix something in 1.7 again LOL. I guess I have to weigh my options ...
> 
> Oooooorrrrrrr, just stop using legacy b.s. and let it die.

Yeah, if only. I love the new versions so much.
